File: scripts/taskupload_client.py
#!/usr/bin/env python
import sys
import rospy
from geometry_msgs.msg import Point
from tsu200.srv import TaskUpload
from tsu200.msg import Pos


def call_taskupload(id_val, home, positions):
    rospy.wait_for_service('task_upload')
    try:
        svc = rospy.ServiceProxy('task_upload', TaskUpload)
        resp = svc(id_val, home, len(positions), positions)
        return resp
    except rospy.ServiceException as e:
        rospy.logerr('Service call failed: %s', e)
        return None


if __name__ == '__main__':
    rospy.init_node('taskupload_client')
    # Build a default HomePos and positions if none provided
    home = Point(x=0.0, y=0.0, z=0.0)
    pos_list = []
    # Example: create two positions
    p1 = Pos(x=1.0, y=1.0, z=0.0, task_type='takeoff', info='first')
    p2 = Pos(x=2.0, y=2.0, z=1.5, task_type='survey', info='second')
    pos_list.append(p1)
    pos_list.append(p2)

    # Allow overriding id from argv
    id_val = int(sys.argv[1]) if len(sys.argv) > 1 else 1
    resp = call_taskupload(id_val, home, pos_list)
    if resp:
        print('id:', resp.id)
        print('success:', resp.success)
        print('status_code:', resp.status_code)
        print('message:', resp.message)

File: src/taskupload_client.cpp
#include <ros/ros.h>
#include "tsu200/TaskUpload.h"
#include "tsu200/Pos.h"
#include <geometry_msgs/Point.h>

int main(int argc, char **argv) {
  ros::init(argc, argv, "taskupload_client_cpp");
  ros::NodeHandle nh;
  ros::ServiceClient client = nh.serviceClient<tsu200::TaskUpload>("task_upload");
  tsu200::TaskUpload srv;

  int id = 1;
  if (argc > 1) id = atoi(argv[1]);

  geometry_msgs::Point home;
  home.x = 0.0; home.y = 0.0; home.z = 0.0;
  srv.request.id = id;
  srv.request.HomePos = home;

  tsu200::Pos p1; p1.x = 1.0; p1.y = 1.0; p1.z = 0.0; p1.task_type = "takeoff"; p1.info = "first";
  tsu200::Pos p2; p2.x = 2.0; p2.y = 2.0; p2.z = 1.5; p2.task_type = "survey"; p2.info = "second";
  srv.request.PosList.push_back(p1);
  srv.request.PosList.push_back(p2);
  srv.request.PosNum = srv.request.PosList.size();

  if (client.call(srv)) {
    ROS_INFO("id=%d success=%s status_code=%d message=%s", srv.response.id, srv.response.success?"true":"false", srv.response.status_code, srv.response.message.c_str());
  } else {
    ROS_ERROR("Failed to call service task_upload");
    return 1;
  }
  return 0;
}
